File: core/src/com/mygdx/game/deckengine/cards/StrikeCard.java
package com.mygdx.game.deckengine.cards;

import com.badlogic.gdx.graphics.Texture;
import com.mygdx.game.character.CharacterStats;
import com.mygdx.game.character.CharacterType;

public class StrikeCard extends Card {

    public StrikeCard(Texture img) {
        super(img);
    }

    @Override
    public boolean CanApplyEffects(CharacterStats stats) {

        return (stats.getCharacterType() == CharacterType.ENEMY);
    }

    @Override
    public void ApplyEffects(CharacterStats stats) {
        stats.AdjustHealth(-6);
    }
}

File: deck/src/classicthunder/card/impl/StrikeCard.kt
package classicthunder.card.impl

import classicthunder.card.Card
import classicthunder.character.CharacterStats
import classicthunder.character.CharacterType

class StrikeCard(parent: Card?) : Card(parent) {

    private var damage: Int

    init {
        damage = 6
    }

    override fun getEnergyCost(): Int {
        return 1
    }

    override fun canPlay(stats: CharacterStats): Boolean {
        return stats.characterType == CharacterType.ENEMY
    }

    override fun play(stats: CharacterStats) {
        stats.dealDamage(damage)
    }

    override fun copy(parent: Card): StrikeCard {
        val copy = StrikeCard(parent)
        copy.damage = this.damage
        return copy
    }
}